lib/utils/prompt/context_reader_test.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prompt
+
+import (
+	"context"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/term"
+)
+
+func TestContextReader(t *testing.T) {
+	pr, pw := io.Pipe()
+	t.Cleanup(func() { pr.Close() })
+	t.Cleanup(func() { pw.Close() })
+
+	write := func(t *testing.T, s string) {
+		_, err := pw.Write([]byte(s))
```

Comment:
```suggestion
	write := func(t *testing.T, s string) {
		t.Helper()
		_, err := pw.Write([]byte(s))
```

Just for nicer failure messages. (same goes for below)